handlers/token.go

```diff
@@ -0,0 +1,121 @@
+package handlers
+
+import (
+	"github.com/dgrijalva/jwt-go"
+	"github.com/go-chi/jwtauth"
+	"log"
+	"net/http"
+	"time"
+)
+
+var Secret = []byte("codingislife")
+
+var TokenAuth *jwtauth.JWTAuth
+
+func init() {
+	// create token
+	TokenAuth = jwtauth.New("HS256", Secret, nil)
+}
+
+func CreateJWT(username string) (string, error) {
+	// create claims
+	claims := jwt.MapClaims{
+		"username":   username,
+		"authorized": true,
+		"exp":        time.Now().Add(time.Hour * 10).Unix(),
+	}
+
+	// Encode the claims with TokenAuth
+	_, tokenString, err := TokenAuth.Encode(claims)
+	if err != nil {
+		log.Println(err)
+		return "", err
+	}
+
+	return tokenString, nil
+}
+
+func VerifyJWT(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		token, claims, err := jwtauth.FromContext(r.Context())
+		if err != nil || token == nil || !token.Valid {
+			log.Println("someone tried to access protected method without authentication")
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte("authorization information missing/invalid"))
+			return
+		}
+
+		// verify the claims
+		if claims["username"] == nil || claims["authorized"] == nil || claims["exp"] == nil {
+			log.Println("invalid claims")
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte("invalid claims"))
+			return
+		}
+
+		// check username data type
+		username, ok := claims["username"].(string)
+		if !ok {
+			log.Println("username claim is not a string")
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte("username claim is not a string"))
```

Comment:
you don't have to return the logic of your program.
```suggestion
			w.Write([]byte("unauthorized"))
```